textdb/textdb-exp/src/main/java/edu/uci/ics/textdb/exp/nlp/splitter/NlpSplitOperator.java

```diff
@@ -0,0 +1,181 @@
+    }
+    
+    /*
+     * adds a new field to the schema, with name resultAttributeName and type list of strings
+     */
+    private Schema transformSchema(Schema inputSchema) {
+        if (inputSchema.containsField(predicate.getResultAttributeName())) {
+            throw new RuntimeException(String.format(
+                    "result attribute name %s is already in the original schema %s", 
+                    predicate.getResultAttributeName(),
+                    inputSchema.getAttributeNames()));
+        }
+        return Utils.addAttributeToSchema(inputSchema, 
+                new Attribute(predicate.getResultAttributeName(), AttributeType.LIST));
+    }
+      
+
+    @Override
+    public void open() throws TextDBException {
+        if (cursor != CLOSED) {
+            return;
+        }
+        if (inputOperator == null) {
+            throw new DataFlowException(ErrorMessages.INPUT_OPERATOR_NOT_SPECIFIED);
+        }
+        inputOperator.open();
+        Schema inputSchema = inputOperator.getOutputSchema();
+        
+        // check if input schema is present
+        if (! inputSchema.containsField(predicate.getInputAttributeName())) {
+            throw new RuntimeException(String.format(
+                    "input attribute %s is not in the input schema %s",
+                    predicate.getInputAttributeName(),
+                    inputSchema.getAttributeNames()));
+        }
+        
+        // check if attribute type is valid
+        AttributeType inputAttributeType = 
+                //Does predicate give the correct input attribute name?
+                inputSchema.getAttribute(predicate.getInputAttributeName()).getAttributeType();
+        boolean isValidType = inputAttributeType.equals(AttributeType.STRING) || 
+                inputAttributeType.equals(AttributeType.TEXT);
+        if (! isValidType) {
+            throw new RuntimeException(String.format(
+                    "input attribute %s must have type String or Text, its actual type is %s",
+                    predicate.getInputAttributeName(),
+                    inputAttributeType));
+        }
+        
+        // generate output schema by transforming the input schema
+        outputSchema = transformSchema(inputOperator.getOutputSchema());
+        
+        cursor = OPENED;
+
+    }
+
+    @Override
+    public Tuple getNextTuple() throws TextDBException {
+        if (cursor == CLOSED) {
+            return null;
+        }
+        
+        List<IField> outputFields = new ArrayList<>();
+        
+        if(predicate.getOutputType() == NLPOutputType.ONE_TO_ONE) {
+            currentTuple = inputOperator.getNextTuple();
+            if (currentTuple == null) {
+                return null;
+            }
+            currentSentenceList = getSentenceList(currentTuple);
+            outputFields.addAll(currentTuple.getFields());
+            outputFields.add(new ListField<TextField>(currentSentenceList));
```

Comment:
for oneToOne, the output field you added should be a `ListField<Span>`. Although technically you can have a `ListField<TextField>`, such a field is never supported elsewhere in the engine. So please change it to a `ListField<Span>`